src/taipy/core/notification/core_event_consumer.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2023 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+import abc
+import threading
+from queue import Empty, SimpleQueue
+
+from .event import Event
+
+
+class CoreEventConsumer(threading.Thread):
```

Comment:
neat! I think this example is really great, so we will only need to retrieve the register_id and the queue from the notifier to instantiate the consumer, which IMO is simple and straightforward to implement 😃 I agree this class can be a parent/abstract class so that we can hide away the thread manipulation part (can be a pain from time to time) and only exposing the process_event (or the business logic)!